crates/bevy_reflect/src/enums/dynamic_enum.rs

```diff
@@ -140,6 +140,14 @@ impl DynamicEnum {
         self.variant = variant.into();
     }
 
+    pub fn variant(&self) -> &DynamicVariant {
+        &self.variant
+    }
+
+    pub fn variant_mut(&mut self) -> &mut DynamicVariant {
+        &mut self.variant
+    }
```

Comment:
This could allow for users to update the variant without syncing the variant name or index. If we want this method, could we at least add some documentation letting users know they should take care not to change the variant through this method? Probably also with a link to the two setter methods?